server/protocols/webadmin/webadmin-mailrepository/src/test/java/org/apache/james/webadmin/service/ReprocessingAllMailsTaskTest.java

```diff
@@ -0,0 +1,93 @@
+import org.junit.jupiter.params.provider.MethodSource;
+import org.junit.jupiter.params.provider.ValueSource;
+
+class ReprocessingAllMailsTaskTest {
+    private static final ReprocessingService REPROCESSING_SERVICE = mock(ReprocessingService.class);
+
+    @ParameterizedTest
+    @MethodSource
+    void taskShouldBeSerializable(long repositorySize,
+                                  MailRepositoryPath repositoryPath,
+                                  String targetQueue,
+                                  Optional<String> targetProcessor,
+                                  String serialized) throws JsonProcessingException {
+        JsonTaskSerializer testee = new JsonTaskSerializer(ReprocessingAllMailsTask.MODULE.apply(REPROCESSING_SERVICE));
+        ReprocessingAllMailsTask task = new ReprocessingAllMailsTask(REPROCESSING_SERVICE, repositorySize, repositoryPath, targetQueue, targetProcessor);
+        JsonAssertions.assertThatJson(testee.serialize(task))
+            .isEqualTo(serialized);
+    }
+
+    private static Stream<Arguments> taskShouldBeSerializable() {
+        return allValidTasks();
+    }
+
+    @ParameterizedTest
+    @MethodSource
+    void taskShouldBeDeserializable(long repositorySize,
+                                    MailRepositoryPath repositoryPath,
+                                    String targetQueue,
+                                    Optional<String> targetProcessor,
+                                    String serialized) throws IOException {
+        JsonTaskSerializer testee = new JsonTaskSerializer(ReprocessingAllMailsTask.MODULE.apply(REPROCESSING_SERVICE));
+        ReprocessingAllMailsTask task = new ReprocessingAllMailsTask(REPROCESSING_SERVICE, repositorySize, repositoryPath, targetQueue, targetProcessor);
+
+        assertThat(testee.deserialize(serialized))
+            .isEqualToComparingFieldByFieldRecursively(task);
+    }
+
+    private static Stream<Arguments> taskShouldBeDeserializable() {
+        return allValidTasks();
+    }
+
+    private static Stream<Arguments> allValidTasks() {
+        return Stream.of(
+            Arguments.of(5L, MailRepositoryPath.from("a"), "queue", Optional.of("targetProcessor"), "{\"type\":\"reprocessingAllTask\",\"repositorySize\":5,\"repositoryPath\":\"a\",\"targetQueue\":\"queue\",\"targetProcessor\":\"targetProcessor\"}"),
+            Arguments.of(5L, MailRepositoryPath.from("a"), "queue", Optional.empty(), "{\"type\":\"reprocessingAllTask\",\"repositorySize\":5,\"repositoryPath\":\"a\",\"targetQueue\":\"queue\"}")
+        );
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"{\"type\":\"reprocessingAllTask\",\"repositorySize\":5,\"repositoryPath\":\"%\",\"targetQueue\":\"queue\",\"targetProcessor\":\"targetProcessor\"}", "{\"type\":\"reprocessingAllTask\",\"repositorySize\":5,\"repositoryPath\":\"%\",\"targetQueue\":\"queue\"}"})
+    void taskShouldThrowOnDeserializationUrlDecodingError(String serialized) {
```

Comment:
you are not testing UrlEncodingFailureSerializationException (and the test name looks wrong)

Reply:
I did not find any way to trigger it (this Exception is thrown when the Charset in wrong).